### PR TITLE
[API] Add __call__ method

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This PR is a bit dependent on: #3 

I think it could make sense to add a `__call__` method that also allows to take a list as an input and always returns a tensor of shape [bsz, seq_len, feature_dim] as it's usually done in ML IMO.

Also, I've added a `torch.no_grad()` decorator here so that gradients are never stored with this method to save memory. Think this makes sense as I don't really see prompt weighting to be used for training. 

Happy to write tests for it, but was wondering if @damian0815 could say before if this PR is desired or not.

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
With this PR a common use case being:

```py
prompt_embeds = torch.cat([compel.build_conditioning_tensor(prompt) for prompt in prompts])
```

could be reduced to just:
```py
prompt_embeds = compel(...)
```

**Review by damian0815 — APPROVED** [review]
